### PR TITLE
feat(TDP-4712): user pref

### PR DESCRIPTION
**Pull request by frassinier**
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-4712

**Please check if the PR fulfills these requirements**
- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [x] The new code does not introduce new technical issues (sonar / eslint)
- [x] Functional tests have been performed
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

**Please check the browsers you've tested on**
- [ ] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [x] No, and no need to (backend changes only)

### CODE REVIEW

File: dataprep-backend-service/src/main/java/org/talend/dataprep/security/NoOpSecurity.java [hunk 1]
```diff
@@ -67,4 +68,9 @@ public boolean isTDPUser() {
         return true;
     }
 
+    @Override
+    public Locale getLocale() {
```
Comment (hmdebenque) [inline dataprep-backend-service/src/main/java/org/talend/dataprep/security/NoOpSecurity.java]: Why not use the `LocaleContextResolver` from Spring ?

File: dataprep-backend-service/src/test/java/org/talend/dataprep/i18n/DataprepLocaleContextResolverTest.java [hunk 2]
```diff
@@ -41,32 +43,52 @@
...
-        assertEquals(Locale.US, localeContext.getLocale());
+        Locale locale = resolver.resolveLocale(request);
+        assertNotNull(locale);
+        assertEquals(Locale.US, locale);
+    }
+
+    @Test
+    public void resolveLocaleContext_fromHttpHeader() throws Exception {
+        DataprepLocaleContextResolver resolver = new DataprepLocaleContextResolver(null);
+        when(request.getHeader(eq("Accept-Language"))).thenReturn(Locale.JAPANESE.toLanguageTag());
```
[the remark this answers is not in the document]
Comment (fhuaulme) [reply dataprep-backend-service/src/test/java/org/talend/dataprep/i18n/DataprepLocaleContextResolverTest.java]: 👍

File: dataprep-backend-service/src/main/java/org/talend/dataprep/security/NoOpSecurity.java (hunk 1 above)
Comment (fhuaulme) [reply dataprep-backend-service/src/main/java/org/talend/dataprep/security/NoOpSecurity.java]: 👍 (if you meant `org.springframework.context.i18n.LocaleContextHolder#getLocale`)

File: dataprep-backend-service/src/main/java/org/talend/dataprep/i18n/DataprepLocaleContextResolver.java [hunk 3]
```diff
@@ -45,13 +44,13 @@ public DataprepLocaleContextResolver(@Value("${dataprep.locale:en-US}") String c
     }
 
     @Override
-    public LocaleContext resolveLocaleContext(HttpServletRequest request) {
-        return new SimpleLocaleContext(applicationLocale);
+    public Locale getDefaultLocale() {
```
[the remark this answers is not in the document]
Comment (fhuaulme) [reply dataprep-backend-service/src/main/java/org/talend/dataprep/i18n/DataprepLocaleContextResolver.java]: indeed 👍

File: dataprep-backend-service/src/main/java/org/talend/dataprep/i18n/DataprepLocaleContextResolver.java [hunk 4]
```diff
@@ -15,28 +15,27 @@
...
-import org.springframework.context.i18n.SimpleLocaleContext;
 import org.springframework.stereotype.Component;
-import org.springframework.web.servlet.i18n.AbstractLocaleContextResolver;
+import org.springframework.web.servlet.i18n.AcceptHeaderLocaleResolver;
 
 // This component must have the LOCALE_RESOLVER_BEAN_NAME as it is searched by this name in
 // DispatcherServlet.initLocaleResolver:524
 @Component(LOCALE_RESOLVER_BEAN_NAME)
-public class DataprepLocaleContextResolver extends AbstractLocaleContextResolver {
+public class DataprepLocaleContextResolver extends AcceptHeaderLocaleResolver {
```
[the remark this answers is not in the document]
Comment (fhuaulme) [reply dataprep-backend-service/src/main/java/org/talend/dataprep/i18n/DataprepLocaleContextResolver.java]: 👍

File: dataprep-backend-service/src/main/java/org/talend/dataprep/i18n/DataprepLocaleContextResolver.java [hunk 5]
```diff
@@ -62,17 +61,17 @@ private Locale resolveApplicationLocale(String configuredLocale) {
                 if (LocaleUtils.isAvailableLocale(locale)) {
                     LOGGER.debug("Setting application locale to configured {}", locale);
                 } else {
-                    locale = getDefaultLocale();
+                    locale = DEFAULT_LOCALE;
```
[the remark this answers is not in the document]
Comment (fhuaulme) [reply dataprep-backend-service/src/main/java/org/talend/dataprep/i18n/DataprepLocaleContextResolver.java]: Added javadoc

File: dataprep-backend-service/src/main/java/org/talend/dataprep/i18n/DataprepLocaleContextResolver.java [hunk 6]
```diff
@@ -62,20 +81,31 @@ private Locale resolveApplicationLocale(String configuredLocale) {
...
                 } else {
-                    locale = getDefaultLocale();
-                    LOGGER.debug("Locale {} is not available. Defaulting to {}", configuredLocale, getDefaultLocale());
+                    locale = DEFAULT_LOCALE;
+                    LOGGER.debug("Locale {} is not available. Defaulting to {}", configuredLocale,
+                            delegate.getDefaultLocale());
                 }
             } catch (IllformedLocaleException e) {
-                locale = getDefaultLocale();
+                locale = DEFAULT_LOCALE;
```
Comment (hmdebenque) [inline dataprep-backend-service/src/main/java/org/talend/dataprep/i18n/DataprepLocaleContextResolver.java]: DEFAULT_LOCALE => delegate.getDefaultLocale()

Comment (fhuaulme) [reply dataprep-backend-service/src/main/java/org/talend/dataprep/i18n/DataprepLocaleContextResolver.java]: 👎

Comment (hmdebenque) [reply dataprep-backend-service/src/main/java/org/talend/dataprep/i18n/DataprepLocaleContextResolver.java]: :'(

Comment (hmdebenque) [reply dataprep-backend-service/src/main/java/org/talend/dataprep/i18n/DataprepLocaleContextResolver.java]: you're right

**Pull request merged**